cmd/gpu-allocator/README.md

```diff
@@ -0,0 +1,49 @@
+Build and push gpu-allocator container image (use your favorate `ALLOCATOR_IMG_REG`).
+```shell
+make build-gpu-allocator ALLOCATOR_IMG_REG=quay.io/junatibm
+make push-gpu-allocator ALLOCATOR_IMG_REG=quay.io/junatibm
+```
+
+Create a pod to test the gpu-allocator container.
+```shell
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Pod
+metadata:
+  name: my-server-request
+  labels:
+    app: my-server-request
+spec:
+  containers:
+    - name: gpu-allocator
+      image: quay.io/junatibm/gpu-allocator:latest
+      imagePullPolicy: Always
+      ports:
+        - containerPort: 8080
+      readinessProbe:
+        httpGet:
+          path: /readyz
+          port: 8080
+        initialDelaySeconds: 2
+        periodSeconds: 5
+        failureThreshold: 3
+      resources:
+        limits:
+          nvidia.com/gpu: "1"
+          cpu: "1"
+          memory: 250Mi
+EOF
+```
+
+Check the readiness and exposed GPU.
+```console
+(vllm) ubuntu@ip-172-31-58-228:~/llm-d-fast-model-actuation$ kubectl get po my-server-request -owide
+NAME                READY   STATUS    RESTARTS   AGE   IP           NODE               NOMINATED NODE   READINESS GATES
+my-server-request   1/1     Running   0          57s   10.0.0.106   ip-172-31-58-228   <none>           <none>
+(vllm) ubuntu@ip-172-31-58-228:~/llm-d-fast-model-actuation$ thepodip=10.0.0.106
+(vllm) ubuntu@ip-172-31-58-228:~/llm-d-fast-model-actuation$ curl $thepodip:8080/readyz
+ready(vllm) ubuntu@ip-172-31-58-228:~/llm-d-fast-model-actuation$
+(vllm) ubuntu@ip-172-31-58-228:~/llm-d-fast-model-actuation$ curl $thepodip:8080/dual-pod/accelerators
+["GPU-b88683d2-a275-15af-5633-d569397bd622"]
```

Comment:
We will need the answer in terms that can be put in `CUDA_VISIBLE_DEVICES`. Does vLLM succeed when GPU UUIDs are put there?

Reply:
it should: https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#env-vars

Reply:
Shouldn't we use `NVIDIA_VISIBLE_DEVICES` here instead of `CUDA_VISIBLE_DEVICES`?

Reply:
@waltforme: no. You and I have experimentally confirmed that NVIDIA_VISIBLE_DEVICES is not a control knob that a PodSpec can usefully set.

Reply:
Our dual-pods technique is special, because
1. By design, the server-runinng pod will bypass the nvidia device plugin, making the user-specified `NVIDIA_VISIBLE_DEVICES` the exactly one instance (instead of two instances) of the envar.
2. Since the user-specified `NVIDIA_VISIBLE_DEVICES` takes precedence over the vLLM image's baked-in `NVIDIA_VISIBLE_DEVICES=all`, the vLLM container will respect the former, and sees precisely what specifed by the user instead of all GPUs.

Therefore, `NVIDIA_VISIBLE_DEVICES` _is_ a precise control knob for the dual-pods use case.

Reply:
Oh, right, when requesting zero GPUs, the use's specified NVIDIA_VISIBLE_DEVICES is passed through.

Regardless of which variable is used, the more important question here is whether GPU UUIDs are consumable by vLLM. See the earlier comments in this thread.